src/components/Filter/Filter.jsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react'
+import { useProduct } from '../../context';
+import './Filter.css';
+
+
+
+const Filter = () => {
+
+  const { productDispatch } = useProduct();
+
+  return (
+    <div>
+      <aside className="sidebar">
+        <div className="space-between">
+          <p className="text-l"><b>Filter</b></p>
+          <button className='clear-btn' onClick={()=>productDispatch({type:"CLEAR"})}>Clear</button>
+        </div>
+
+        <div className="filter-group">
+          <p className="text-m"><b>Sort</b></p>
+          <div className="filter-item">
+            <label>
+              <input type="radio" name="price" id="filter-price" 
+              onChange={() => productDispatch({ type: "SORT", payload: "LOW_TO_HIGH" })} />
+              Price (Low to High)</label>
+          </div>
+          <div className="filter-item">
+            <label>
+              <input type="radio" name="price" id="filter-price" 
+              onChange={() => productDispatch({ type: "SORT", payload: "HIGH_TO_LOW" })} />
+              Price (High to Low)</label>
+          </div>
+        </div>
+        <div className="filter-group">
+          <p className="text-m"><b>Metal</b></p>
+          <div className="filter-item">
+            <label>
+              <input type="checkbox" name="metal" id="filter-metal" 
+              onChange={()=>productDispatch({type:"GOLD"})}/>
+              Gold</label>
+          </div>
+          <div className="filter-item">
+            <label>
+              <input type="checkbox" name="metal" id="filter-metal" 
+              onChange={()=>productDispatch({type:"SILVER"})}/>
+              Silver</label>
+          </div>
+          <div className="filter-item">
+            <label>
+              <input type="checkbox" name="metal" id="filter-metal" onChange={()=>productDispatch({type:"PLATINUM"})}/>
+              Platinum</label>
+          </div>
+        </div>
+        <div className="filter-group">
+          <p className="text-m"><b>Category</b></p>
```

Comment:
in sort by category instead of calling dipsatch to each category you can add selected category to array or object and you can sort on the basis of that.